[MC-40] Create Component Logic Password Input

pedroemanuel42 opened the pull request:
# Feature/Issue Description

This PR create a class password components that has get/set values, validations, required and disabled checks with tests around then.

# Changes Summary

- Add get/set Value;
- Add password input validations (must be same as input, at least size 8, contain at least 1 capital letter, 1 number, and 1 special character);
- Add get/set valid, enabled and disabled;
- Add verify into valid, enabled and disabled;
- Add tests to all this new methods.

# How Has This Been Tested?

To test all methods, i created new obects using them, checked in the localhost log to see these returns.

pedroemanuel42 committed:
MC-40 - Created PasswordInputComponents

pedroemanuel42 committed:
MC-40 - Fixed input-password.components.test.js

pedroemanuel42 committed:
MC-40 - Separeted PasswordInputComponents methods and fixed test

pedroemanuel42 committed:
MC-40 Changed toBe(true) for toBeTruthy() in PassowrdInputComponents tests

pedroemanuel42 committed:
MC-40 Fixed lint problems

pedroemanuel42 committed:
MC-40 Build password input requirements

pedroemanuel42 committed:
MC-40 Re-positioned div and forms passwords

pedroemanuel42 committed:
MC-40 added display none as comment

pedroemanuel42 committed:
MC-40 Added input password validations

-Added input password validations using regex;
-Refactored password  html and js components.

pedroemanuel42 committed:
MC-40 Added button query, changed checkQuery and verifyPswd

- Also removed the input type check for button and password.

pedroemanuel42 committed:
MC-40 Refactored password container and password js component

pedroemanuel42 committed:
MC-40 Refactored password component and test

pedroemanuel42 committed:
MC-40 Refactored password component

pedroemanuel42 committed:
MC-40 Rename password component test

pedroemanuel42 committed:
MC-40 Lint regex

pedroemanuel42 committed:
MC-40 Added IsEnable & isRequired

pedroemanuel42 committed:
MC-40 refactored code

pedroemanuel42 committed:
MC40- Added sets and gets to methods

pedroemanuel42 committed:
MC-40 Added input type to query's  icon/input

pedroemanuel42 committed:
MC-40 Refactored valid,  required and disabled

pedroemanuel42 committed:
MC-40 Added tests to PassowrdInput methods

pedroemanuel42 committed:
MC-40 Adjusted password test component

pedroemanuel42 committed:
Merge branch 'develop' into feature/MC-40

pedroemanuel42 committed:
MC-40 Refactored my code after merge develop

pedroemanuel42 committed:
MC-40 run lint and changed isDisabled return

pedroemanuel42 committed:
MC-40 npm install to solve run test problem

tvSantiago commented:
Your description of how has this been tested is confusing.
First don't possible how to know test. Must create a new object? Or just run the server?

tvSantiago commented on src/assets/scss/_variables.scss:
```diff
@@ -28,4 +28,4 @@ $primary-light: #474787;
 $other-dark: #d1ccc0;
-$other-light: #ffffff;
\ No newline at end of file
+$other-light: #ffffff;
```
Must have an empty line in the file end

tvps20 replied:
![image](https://user-images.githubusercontent.com/23269934/162721505-e40dedd2-6c5f-4a34-ba34-cade7ba24596.png)

tvSantiago commented on src/assets/scss/dropdown.scss:
```diff
@@ -87,4 +87,3 @@
     pointer-events: none;
     color: $other-dark;
 }
-
```
Must have an empty line in the file end

tvSantiago commented on src/assets/scss/home-layout.scss:
```diff
@@ -35,4 +35,3 @@ $nav-padding: 10px;
     display: flex;
     padding: 10px 10%;
 }
-
```
Must have an empty line in the file end

tvSantiago commented on src/assets/scss/input-password.scss:
```diff
@@ -31,18 +30,18 @@
-    left: -17px; 
+    top: -32px;
+    left: -17px;
 }
```
Must have an empty line in the file end

tvSantiago commented on src/components/password.components.js:
```diff
@@ -0,0 +1,98 @@
+        return this.input.prop('disabled');
+    }
+
+    setDisabled(boolean) {
```
This function when passing a value true changes the style:
![image](https://user-images.githubusercontent.com/92385147/162186991-0e702b1a-d7d3-44c8-a954-1422ae1c11eb.png)
I can also click on the icon to show or hide a password:
![image](https://user-images.githubusercontent.com/92385147/162187161-69769cc8-238c-46d7-9e37-35da99883d32.png)

tvSantiago commented on src/components/password.components.js:
```diff
@@ -0,0 +1,98 @@
+        this.input.val(string);
+    }
+
+    isValid() {
```
There's a logical problem here
![image](https://user-images.githubusercontent.com/92385147/162188972-8e107b4e-8fb0-452b-81db-92764ac73dad.png)

tvSantiago commented on src/components/password.components.js:
```diff
@@ -0,0 +1,98 @@
+        this.disabled = boolean;
+    }
+
+    isDisabled() {
```
This is redundant, this is also for the other properties
![image](https://user-images.githubusercontent.com/92385147/162190982-b6d8081b-05dd-43c8-ab0b-f7c6a8856416.png)

tvps20 replied:
If I click on any point of the component it activates the show/hide should only occur by clicking on the icon
![image](https://user-images.githubusercontent.com/23269934/162725523-02977bcc-3956-43e7-b576-cc5790be5308.png)

tvSantiago commented on src/components/password.components.js:
```diff
@@ -0,0 +1,98 @@
+        this.initFor();
+    }
+
+    hasQuery(div, input, icon) {
```
These names do not reflect what the functions are doing
![image](https://user-images.githubusercontent.com/92385147/162191812-f477530f-b6fd-4271-9399-c82ee6102aea.png)

pedroemanuel42 committed:
MC-40 disabled CSS fix

pedroemanuel42 committed:
MC-40 Review changes on PR

- Changed Query check name;
-Removed redundation on checks properties;
- Resolved logic problems on checks properties;
- Also added blank space line on items (last commit);

pedroemanuel42 committed:
MC-40 Changed check query test name

pedroemanuel42 commented:
Solved, waiting review again. Thanks for the advices!

pedroemanuel42 commented:
> Your description of how has this been tested is confusing. First don't possible how to know test. Must create a new object? Or just run the server?

Edited, thanks!

pedroemanuel42 committed:
MC-40 Solved logic problems

tvps20 commented on src/pages/components/style.scss:
```diff
@@ -59,4 +60,3 @@ p {
     background-color: $primary-light;
     color: $other-light;
 }
```
Must have an empty line in the file end

tvps20 commented on src/pages/components/main.js:
```diff
@@ -1,13 +1,14 @@
     // eslint-disable-next-line no-unused-vars
     const dropdownGameType = new DropdownComponent('#games');
-    PasswordInputService.initFor('#password');
 });
```
Must have an empty line in the file end

tvps20 commented on src/components/tests/password.components.test.js:
```diff
@@ -0,0 +1,254 @@
+            expect(password.isRequired()).toBeTruthy();
+        });
+    });
+});
```
Must have an empty line in the file end

feelsantiago commented on src/components/password.components.js:
```diff
@@ -0,0 +1,89 @@
+        this.input.val(string);
+    }
+
+    isValid() {
```
You guys must use the same name for the validation method. And must have the same signature and return.

feelsantiago commented on src/components/tests/password.components.test.js:
```diff
@@ -0,0 +1,254 @@
+        });
+    });
+
+    describe('[isProperty]', () => {
```
what is isProperty?

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,89 @@
+        }
+
+        return false;
+    }
```
The state is not save
![image](https://user-images.githubusercontent.com/23269934/162724207-9c662a5e-ddb2-4595-b94c-a0c660a1d5e7.png)

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,89 @@
+        }
+    }
+
+    onClick() {
```
This name does not reflect the comportment

pedroemanuel42 committed:
MC-40 Review changes by SM and PM

tvps20 commented:
Pr title is wrong
![image](https://user-images.githubusercontent.com/23269934/163166568-bb9948c8-cb8e-4770-9725-9f1902969d30.png)

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,97 @@
+
+        this.valid = valid;
+        this.disabled = disabled;
+        this.required = required;
```
I don't get object value
![image](https://user-images.githubusercontent.com/23269934/163168225-813fc492-550b-4d2b-96cb-c2fa0fe3e0af.png)

tvps20 replied:
These are states I want, other information can be private.

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,97 @@
+        }
+    }
+
+    addClickEvent() {
```
I don't have methods for these behaviors
![image](https://user-images.githubusercontent.com/23269934/163170330-43a32769-c0ad-47f3-bbeb-8807eb6f636a.png)

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,97 @@
+        return this.input.val();
+    }
+
+    setValue(string) {
```
This comportament is wrong
![image](https://user-images.githubusercontent.com/23269934/163169947-68aa9a5f-75e3-420e-b462-e7a31ccec601.png)

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,97 @@
+        });
+    }
+
+    getValue() {
```
These are types of the comportaments want, other information can be private.
![image](https://user-images.githubusercontent.com/23269934/163170698-160a9457-63ac-4d06-b27e-1e9ab1125d00.png)

EvertonDuarteGuimaraes commented on src/assets/scss/input-password.scss:
```diff
@@ -7,22 +7,21 @@
 
 // Input Password => placeholder => 'Senha'
-.password-input, .password-input:focus {
+.password-input, .password-input:focus, .password-input:disabled {
```
Maybe it's more readable if you try something like:
.password {
....&-input, &-input:focus, &-input:disabled {
........ // styles
....}
}
That's take anything with follows password-input:'something'.

pedroemanuel42 replied:
Did it, thanks for the tips!

pedroemanuel42 committed:
MC-40 Requested changes on password class

pedroemanuel42 committed:
MC-40 refactored validate test

pedroemanuel42 committed:
MC-40 Updated eslint rules

pedroemanuel42 committed:
MC-40 - Added private convention

pedroemanuel42 committed:
MC-40 Modified password tests

pedroemanuel42 committed:
MC-40 Fixed disable method

pedroemanuel42 committed:
MC-40 Added new removeClickEvent method

pedroemanuel42 committed:
MC-40 Changed private states to methods

pedroemanuel42 commented:
> Pr title is wrong ![image](https://user-images.githubusercontent.com/23269934/163166568-bb9948c8-cb8e-4770-9725-9f1902969d30.png)

Fixed.

pedroemanuel42 commented:
> Your description of how has this been tested is confusing. First don't possible how to know test. Must create a new object? Or just run the server?

Fixed.

pedroemanuel42 committed:
MC-40 Removed coments on CSS and added expect test

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,90 @@
+        this.disabled = false;
+
+        this._checkQuery(this._div, this._input, this._icon);
+        this.addClickEvent();
```
this property can private.

tvps20 commented on src/components/password.components.js:
```diff
@@ -0,0 +1,90 @@
+        });
+    }
+
+    removeClickEvent() {
```
why does this behavior exist if disable() already meets the requirements?

tvps20 reviewed (changes requested):
Leave two questions to analyze

feelsantiago commented on src/assets/scss/input-password.scss:
```diff
@@ -3,46 +3,44 @@
-    padding-left: 35px;
-    padding-top: 5px;
-}
+.password- {
```
Remove dash -

feelsantiago commented on src/components/tests/password.components.test.js:
```diff
@@ -0,0 +1,242 @@
+            $(document.body).html(div);
+
+            expect(() => {
+                password = new PasswordInput('#pazz');
```
Don't instantiate global variables in tests. Use beforeEach or BeforeAll methods to setup data. Every test should not affect variables from others tests.

feelsantiago replied:
Use local variables or resete variables on afterEach or afterAll methods

pedroemanuel42 committed:
MC-40 - Requested changes into scss and js

pedroemanuel42 committed:
MC-40 - Requested changes on test assign variables

pedroemanuel42 committed:
Merge branch 'develop' into feature/MC-40

pedroemanuel42 committed:
MC-40 - Added no-unsed-vars lint
